app/src/main/java/com/example/henzoshimada/feeltrip/FeelTripApplication.java

```diff
@@ -59,4 +65,79 @@ public static Participant getParticipant(){
         return participant;
     }
 
+    private static FilterController filterController = null;
+    public static FilterController getFilterController(){
+        if (filterController == null){
+            filterController = new FilterController();
+        }
+        return filterController;
+    }
+
+    private static  MoodAdapter moodAdapter = null;
+    private static ArrayList<Mood> moodArrayList = new ArrayList<Mood>();
+    public static MoodAdapter getMoodAdapter(Context context) {
+        if (moodAdapter == null) {
+            moodAdapter = new MoodAdapter(moodArrayList, context.getApplicationContext());
+        }
+        return moodAdapter;
+    }
+
+    public static ArrayList<Mood> getMoodArrayList() {
+        if(moodArrayList == null) {
+            moodArrayList = new ArrayList<Mood>();
+        }
+        return moodArrayList;
+    }
+
+    public static void setMoodArrayList(ArrayList<Mood> moodArrayList) {
+        FeelTripApplication.moodArrayList = moodArrayList;
+    }
+
+    private static int frag; // 0 for  Home frag, 1 for Profile frag, 2 for Map frag
+    public static void setFrag(String fragstr) {
+        switch(fragstr) {
+            case "main":
+                frag = 0;
```

Comment:
For the purposes of knowing which fragment we're currently viewing: main, profile, or map